nixos/modules/services/misc/cfdyndns.nix

```diff
@@ -55,12 +64,15 @@ in
         Group = config.ids.gids.cfdyndns;
       };
       environment = {
-        CLOUDFLARE_EMAIL="${cfg.email}";
         CLOUDFLARE_RECORDS="${concatStringsSep "," cfg.records}";
       };
       script = ''
         ${optionalString (cfg.apikeyFile != null) ''
           export CLOUDFLARE_APIKEY="$(cat ${escapeShellArg cfg.apikeyFile})"
+          export CLOUDFLARE_EMAIL="${cfg.email}"
+        ''}
+        ${optionalString (cfg.apiTokenFile != null) ''
```

Comment:
What will be used if both `apikeyFile` and `apiTokenFile` are set?  Should be add an assertion to make sure only one of them is set to reduce confusing?

Reply:
Well it depends on the implementation. In the current package in nixpkgs, the Rust binary prefers the api key, email combo, and will use it if both are present. In my recent [refactor](https://github.com/nrdxp/cfdyndns), the inverse is true.

Either way, it won't hurt anything and nothing will break, but perhaps an assert would be a nicety; by no means essential though.

Reply:
This isn't quite true, it will prefer the APITOKEN: https://github.com/colemickens/cfdyndns/blob/31d576a3700989fe7533081b238154289eab84e2/src/main.rs#L42-L55

Reply:
ah did you change that recently or am I just going senile :sweat_smile:

Reply:
I _am_ jetlagged, and I don't want to call you senile... but... according to the blame, it was at least 2 years ago :smile: https://github.com/colemickens/cfdyndns/blame/31d576a3700989fe7533081b238154289eab84e2/src/main.rs#L42-L55

Reply:
I think my question is, why did I implement it in `cfdyndns` and then never make it usable in the module here? Who knows....